hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,208 @@ This document aims to make the distinction between physical entropy and factory
 However, if used unqualified, the term "entropy" should be understood to refer to physical entropy strings which are obtained in accordance with SP 800-90C.
 That is either physical entropy, or the output of a DRNG which itself has been seeded (and possibly reseeded) with physical entropy.
 In CC terms, "entropy strings" (when used in this document without a qualifier) should be understood to come from either a PTG.2 or PTG.3 class RNG.
+
+
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+
+## Block Diagram
+
+
+
+## Hardware Interfaces
+
+
+
+## Design Details
+
+### Application Interface
+
+This section describes the application interface.
+The general interface operates like a FIFO, in which a command is pushed into the interface.
+An optional stream of data may follow, such as seed material for an instantiate command.
+There is one instance of a software interface.
+Many instances of the hardware interface maybe exist, and is determined by the NHwApp parameter.
+
+#### General Command Format
+
+Below is the general format for the application interface commands.
+The following fields are expanded below for clarity.
+
+- `acmd` : application command - selects one of five operations to perform
+- `shid` : state handle identification - hardware uses this to select which csrng maintained state to use
+- `xcnt` : transfer count - sets the total length of the transfer in beats, where zero is 1 beat
+- `flgs` : flags - used to allow for additional features
+- `creq` : count request - used only for the `generate` command, sets the number of genbits bits to be recieved
+
+
+
+                                                
+#### Command Header                                
+The application command header is 32 bits wide. Below is a description of the fields of this header:
+
+```
+  31          23          15            7             0
+  +-----------+-----------+------+------+------+------+
+  +          creq         | flgs | xcnt | shid | acmd |
+  +-----------+-----------+------+------+------+------+
+
+
+```
+
+
+- Total header: bits 31:0
+    - acmd: bits 3-0 
+        - 4'h0: invalid command
+        - 4'h1: instantiate command
+        - 4'h2: reseed command  
+        - 4'h3: generate command
+        - 4'h4: update command
+        - 4'h5: uninstantiate command
+        - 4'h6-4'hF: reserved
+    - shid: bits 7-4
+        - supports up to 16 state handle ids
+    - xcnt: bits 11-8
+        - Number of words (32b/word) to transfer into csrng
+    - flgs: bits 15-11
+        - flags for features
+    - creq: bits 31-16
+        - Number of 128b genbits words (128b/word) requested
+
+
+#### Specific Command Examples
+
+- instantiate:
+    - acmd: 4'h1 - encoded command
+    - shid: 4'h3 - reserved by csrng, set by the app number
+    - xcnt: 4'hc - a count of up to 12 words (32b/word) of seed or additional data 
```

Comment:
Really "up to"?

Reply:
Comment appears out of date with significant revisions.